buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -1884,6 +1884,13 @@ class BeamModulePlugin implements Plugin<Project> {
           }
         }
       }
+      // Set run order for basic tasks.
+      // This should be called after applyPythonNature() since TaskContainer
+      // requires task instances created first before setting the order.
+      project.ext.setTaskOrder = {
```

Comment:
Hm... this feels a bit hacky. setupVirtualEnv is already in `dependsOn` of installGcpTest. 

As for `sdks:python:sdist`, is it possible to add a dependency on `distTarBall` configuration instead?

Reply:
You are right, set `setupVirtualenv`run after `installGcpTest` is not required in here. The main purpose is to set `installGcpTest` run after `sdks:python:sdist` in each project.

I don't know if depend on `distTarBall` will work or not but worth to try, so that we may be able to get rid of `setTaskOrder`

Reply:
Seems we can do `installGcpTest.mustRunAfter configurations.distTarBall`. [This build](https://scans.gradle.com/s/64ylkwsnyt4go) shows correct run order. I'll go ahead and change the code.